### ISSUE TITLE
uwsgi.set_logvar fails in python 3.10

### ISSUE THREAD

**Issue by kuba-lilz** [unbit/uwsgi#2418]
Python 3.10 introduced a breaking change that affects uwsgi.set_logvar

Here's how to reproduce the problem:
- environment: python 3.10 (from docker container `python:3.10.3-slim-bullseye`)
- Flask==2.0.3
- uWSGI==2.0.20

main.py:
```python
import flask
import uwsgi

app = flask.Flask("myapp")

@app.route("/index")
def index():
	uwsgi.set_logvar('foo', 'bar')
	return "OK"


if __name__ == "__main__":

	app.run()
```

uwsgi command to run web server:
`uwsgi --http 127.0.0.1:3000 --module main:app`

Error when accessing index endpoint:
```
[2022-03-24 02:25:06,993] ERROR in app: Exception on /index [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/app/./main.py", line 8, in index
    uwsgi.set_logvar('foo', 'bar')
SystemError: PY_SSIZE_T_CLEAN macro must be defined for '#' formats
[pid: 1996|app: 0|req: 1/1] 127.0.0.1 () {32 vars in 371 bytes} [Thu Mar 24 02:25:06 2022] GET /index => generated 290 bytes in 4 msecs (HTTP/1.1 500) 2 headers in 99 bytes (1 switches on core 0)
```

PY_SSIZE_T_CLEAN error seems to be due to a breaking change introduced in python 3.10:
https://docs.python.org/3/whatsnew/3.10.html#id2

Any chance this can be addressed?
`uwsgi.set_logvar`  is a documented feature ( https://uwsgi-docs.readthedocs.io/en/latest/LogFormat.html#user-defined-logvars ), and according to https://uwsgi-docs.readthedocs.io/en/latest/Changelog-2.0.20.html#uwsgi-2-0-20 uWSGI==2.0.20 officially supports python 3.10

**Comment by Hawkmoon269** [comment]
A am experiencing the same thing with the following set-up: `python-3.10.2` and `uWSGI==2.0.20`

**Comment by CrafterSvK** [comment]
I've used uWSGI 2.1-dev for python 3.10 and it works fine for some time now.

**Comment by rectalogic** [comment]
Using `uwsgi.sharedarea_write()` also triggers this. It's fixed in master https://github.com/unbit/uwsgi/blob/master/plugins/python/uwsgi_python.h#L3 but there has not been a uwsgi release with the fix. We are upgrading to Ubuntu 22.04 and Python 3.10 is the default and uwsgi no longer works. @unbit are there plans to do a new release on pypi with the fix some time soon?
2.0.20 is nearly a year old https://pypi.org/project/uWSGI/2.0.20/

**Comment by xrmx** [comment]
Should be fixed in 2.0.21.
